gradle-jdks-setup/src/main/java/com/palantir/gradle/jdks/setup/CommandRunner.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks.setup;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public final class CommandRunner {
+
+    public static String run(List<String> commandArguments) {
+        try {
+            Process process = new ProcessBuilder().command(commandArguments).start();
+            String output = readAllInput(process.getInputStream());
+            String errorOutput = readAllInput(process.getErrorStream());
```

Comment:
This works if there's not much output, but in general this can result in deadlock. If the process outputs so much error output that it fills up the stderr buffer but has not yet finished, it is blocked until you read from stderr, but this never happens as we're reading from stdout until the process ends. The fix involves reading both concurrently.